runtimevar/blobvar/blobvar.go

```diff
@@ -34,6 +51,90 @@ import (
 	"gocloud.dev/runtimevar/driver"
 )
 
+func init() {
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, &lazyBucketOpener{})
+}
+
+// Scheme is the URL scheme blobvar registers its URLOpener under on runtimevar.DefaultMux.
+const Scheme = "blob"
+
+type lazyBucketOpener struct {
+	mu      sync.Mutex
+	buckets map[string]*blob.Bucket
+}
+
+func (o *lazyBucketOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	bucketURL := u.Query().Get("bucket")
+	if bucketURL == "" {
+		return nil, fmt.Errorf("open variable %q: URL parameter \"bucket\" is required", u)
+	}
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	if o.buckets == nil {
+		o.buckets = map[string]*blob.Bucket{}
+	}
+	bucket := o.buckets[bucketURL]
+	if bucket == nil {
+		var err error
+		bucket, err = blob.OpenBucket(ctx, bucketURL)
+		if err != nil {
+			return nil, fmt.Errorf("open variable %q: OpenBucket failed: %v", u, err)
+		}
+		o.buckets[bucketURL] = bucket
+	}
+	opener := URLOpener{Bucket: bucket}
+	return opener.OpenVariableURL(ctx, u)
+}
+
+// URLOpener opens Variable URLs like "blob://myblobkey?decoder=string".
+// It supports the URL parameters:
+//   - decoder: The decoder to use. Defaults to URLOpener.Decoder, or
+//       runtimevar.BytesDecoder if URLOpener.Decoder is nil.
+//       See runtimevar.DecoderByName for supported values.
+//   - wait: The poll interval; supported values are from time.ParseDuration.
+//       Defaults to 30s.
+type URLOpener struct {
+	// The Bucket to use; required.
+	Bucket *blob.Bucket
```

Comment:
I would assume this `URLOpener` to have a `*blob.URLMux`, not a single `*blob.Bucket`.

Reply:
I made it have both; you can either provide a `Bucket`, in which case it will always use it, or you can provide a `blob.URLMux` and a `bucket` query parameter. The default opener does the latter.